TykTechnologies/tyk-sync
# [TT-849] Bugfix/sync ids

### Description

Fixing sync to retain `api_id` and policies `id`.
Modified the execution order. Now, we sync first the policies and then the API's.

- In order for policy ID matching to work correctly, your Dashboard must have `allow_explicit_policy_id: true` and `enable_duplicate_slugs: true`.
- In order for policy ID matching to work correctly, your Gateway must have `policies.allow_explicit_policy_id: true`.

Invalidates PR #23 .

Docs: https://github.com/TykTechnologies/tyk-docs/pull/1372

Related dashboard PR: https://github.com/TykTechnologies/tyk-analytics/pull/2073

### Testing
Repo with the API+Policy dump: git@github.com:tbuchaillot/tyk-test-sync.git

1. Dump an API + Policy with OIDC configured.
2. Delete them from the dashboard.
3. Use sync to create the policy and API again.
4. Check if a key for that policy can be created.
5. Check if the OIDC token still works for the API.
6. Sync them again. It should update the policy and API instead of deleting and creating new ones.
